python/cuml/test/test_random_forest.py

```diff
@@ -538,28 +538,34 @@ def test_rf_memory_leakage(small_clf, fil_sparse_format, n_iter):
     # Create a handle for the cuml model
     handle, stream = get_handle(use_handle, n_streams=1)
 
+    def get_device_free_memory(index=0):
+        pynvml.nvmlInit()
+        return pynvml.nvmlDeviceGetMemoryInfo(
+            pynvml.nvmlDeviceGetHandleByIndex(index)
+        ).free
```

Comment:
On CI there's potentially multiple jobs running on the same GPU which can make this test flaky. Can we find a different approach?

Reply:
Hmm good point. Also @jrhemstad told me that pynvml may not give accurate result when pool allocator is used.

Reply:
We have been struggling with a better approach to this for awhile. If we used a pool allocator for memory leak tests, would it be possible for RMM to determine how many pointers have been given out and how many have been returned?

Reply:
You could create a logging memory resource and instruct RMM to use it as the default resource for this test, then analyze the log to make sure there's no memory leaking. May be tricky to properly instantiate the MR and clean it up and whatnot in the middle of a test run. cc @shwina who would be the best person to help on that front from the Python side.